cluster/juju/layers/kubernetes-worker/config.yaml

```diff
@@ -58,6 +58,15 @@ options:
 
       The value for this config must be a JSON array of credential objects, like this:
         [{"server": "my.registry", "username": "myUser", "password": "myPass"}]
+  ingress-ssl-chain-completion:
+    type: boolean
+    default: false
+    description: |
+      Enable chain completion for TLS certificates used by the nginx ingress
+      controller.  Set this to true if you would like the ingress controller
+      to attempt auto-retrieval of intermediate certificates.  The default
+      (false) is recommended for all production kubernetes installations, and
+      any environment which does not have outbound Internet access.
```

Comment:
I'm gonna need some context for the `default: false` here. The [nginx-ingress-controller docs](https://kubernetes.github.io/ingress-nginx/user-guide/cli-arguments/) say this defaults to `true`. What's our justification for deviating from that? Who is recommending false for all production Kubernetes installations?

Reply:
@Cynerva  auto-retrieval of intermediate certificates is a fragile process. It can fail for many reasons, for example firewalls blocking traffic or CA website outages. In a secure production environment it requires egress firewall access rules and they might stop working if CAs move IPs.

At the same time we saw nginx-ingress controller pods crash when chain completion fails.

Because of that the Canonical IS decided that disabling completion is the safest setting for production environments. We may end up with incomplete chain but we prefer that to possible outage.

If however you think that charms should have same default as upstream it would still work for us. What we need is a way to disable this functionality. Making it disabled by default is, in our opinion, the best but we would be happy with "true" being default, we can always just set it to "true" in our environments.